OpenRA.Mods.Common/Pathfinder/HierarchicalPathFinder.cs

```diff
@@ -241,6 +252,26 @@ public HierarchicalPathFinder(World world, Locomotor locomotor)
 			// When we build the cost table, it depends on the movement costs of the cells at that time.
 			// When this changes, we must update the cost table.
 			locomotor.CellCostChanged += RequireCostRefreshInCell;
+
+			if (check == BlockedByActor.Immovable)
+			{
+				// When we account for immovable actors, it depends on the actors on the map.
+				// When this changes, we must update the cost table.
+				actorMap.CellUpdated += RequireBlockingRefreshInCell;
+
+				// Determine immovable cells from actors already on the map.
+				cellsWithSuperImmovableActor = actorMap.AllActors()
```

Comment:
`SuperImmovableActor` sounds a bit awkward IMO. How about `StaticActor`?

Reply:
Perhaps ImmobileActor

Reply:
The issue there is that there's potential for confusion with the `Immobile` trait.

Reply:
But Immobile is most correct, as I understand this will include every actor that has `IOccupiesSpace` but doesn't have `IMove` (haven't looked at the code)

Reply:
I knew there would be bikeshedding on this, so here was my line of thought and what I'd like to have and not have in the name:

`BlockedByActor.Immovable` defines what an immovable actor is. The set of actors we need to name is a subset of this. Conceptually it's like a level of BlockedByActor that sits between `None` and `Immovable`. This set of actors is every immovable actor except any actors that are crushable by the current locomotor, are temporary blockers, or are the transit only cells of a building.

Since we are defining a subset of an existing term, I'd quite like if immovable was in the name somewhere. Hence the thought of "super-immovable", or perhaps less fuzzy adjectives like "highly-immovable" or "permanently-immovable"could work.

Alternatively anything that also fulfils the "this name sounds like it would be a subset of 'immovable'" would satisfy me, even if immovable wasn't in the name. If we do use a distinct term, anything that could be confused with terminology elsewhere in the codebase should be avoided though.

Reply:
Would `AbstractPathBlockingActors`, with a comment explaining that these are a subset of non-moving actors, be appropriate?

Reply:
To me `Immovable` implies that the actor can't be moved by other actors, script or any ouside interance, but possibly can move by itself. `Immobile` implies that the unit can't move, period. I think names such as `PermanentImmobleActors` and `immobileActors` sound best. But I'm not aware of `Immobile` trait is and if this naming collision can be solved

Reply:
> Would `AbstractPathBlockingActors`, with a comment explaining that these are a subset of non-moving actors, be appropriate?

I could get behind the idea of using generic names such as `cellsWithBlockingActors`, `ActorIsBlocking`, `ActorCellIsBlocking` and then leaving the comments to explain the policy of what we're considering to be blocking.

Reply:
Updated the naming in code, comments and commit messages to remove references to "super-immovable". Code uses the generic "blocking actors" naming as above. Comments refer to "the subset of immovable actors" when we need to be clear that only some of the actors are considered.